### PR TITLE
1966 tooltip

### PR DESCRIPTION
**Pull request by AlexSofft**

### CODE REVIEW

**Comment by mf-code** [comment]
[5937] - vDppkVO3qVlp

**Comment by mf-code** [comment]
[5937] - h3lvyZRGz6xV

**Comment by mf-code** [comment]
[5937] - t4YzSPnQ76Ne

**Comment by mf-code** [comment]
[5937] - G8Oi8EINGlQ8

**Comment by mf-code** [comment]
[5937] - LNPj8jma9DXx

**Comment by mf-code** [comment]
[5937] - 85rAt1mRRZxH

**Comment by mf-code** [comment]
[5937] - 4Jja4hU3ZHnb

**Comment by mf-code** [comment]
[5937] - 0ONfXPJlIYOW

**Comment by mf-code** [comment]
[5937] - 3lv0qwRMfV3A

**Comment by mf-code** [comment]
[5937] - kTKxTNIyCxG6

**Comment by mf-code** [comment]
[5937] - WLzUV9YhIOo9

**Comment by mf-code** [comment]
[5967] - sfXotc8JbWQp

**Comment by mf-code** [comment]
[5967] - i0saOwNVNQxO

**Comment by mf-code** [comment]
[5967] - qXqfmXZhqxiD

**Comment by mf-code** [comment]
[5967] - 4kvk9QgS4xiw

**Comment by mf-code** [comment]
[5967] - VktHqorv5bCl

**Comment by mf-code** [comment]
[5967] - 2yXskmI8rzC5

**Comment by mf-code** [comment]
[5967] - DA3ZDm92bl00

**Comment by mf-code** [comment]
[5967] - P94ikXAvQhWG

**Comment by mf-code** [comment]
[5967] - ybWQzdMotKNB

**Comment by mf-code** [comment]
[5967] - EI9Yku9Wm9SU

**Comment by mf-code** [comment]
[5967] - rvfyxgOMndou

**Comment by mf-code** [comment]
[5967] - V1ewB8mMsmJv

**Comment by mf-code** [comment]
[5967] - P62YqP3YvUzf

**Comment by mf-code** [comment]
[5967] - I5wNuan8i8zk

**Comment by mf-code** [comment]
[5967] - OSsrHBRrEFhw

**Comment by mf-code** [comment]
[5967] - vHi1UdqlqZH0

**Comment by mf-code** [comment]
[5967] - RQ6PMvVCQ73S

**Comment by mf-code** [comment]
[5967] - openjdk8 - Allure report on Netlify: https://5f4cacdf3025493254748540--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5967] - openjdk10 - Allure report on Netlify: https://5f4cae99a0882984fe7b066f--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5973] - nvd8BJUY7CQc

**Comment by mf-code** [comment]
[5973] - r9IgwfSL5aaF

**Comment by mf-code** [comment]
[5973] - O5c8pnJ73BYY

**Comment by mf-code** [comment]
[5973] - PnYNbeXevWJZ

**Comment by mf-code** [comment]
[5973] - 7Cl6TF1pSt9T

**Comment by mf-code** [comment]
[5973] - uQc1UPAbSBeR

**Comment by mf-code** [comment]
[5973] - 85soJHzp494j

**Comment by mf-code** [comment]
[5973] - KhvFjE6e751I

**Comment by mf-code** [comment]
[5973] - TUSdVh2LXqvj

**Comment by mf-code** [comment]
[5973] - 8vqMOokwQ27Y

**Comment by mf-code** [comment]
[5973] - 830nfk44TTCZ

**Comment by mf-code** [comment]
[5973] - YZdchbY3DfJv

**Comment by mf-code** [comment]
[5973] - t0m4BtvQzmCw

**Comment by mf-code** [comment]
[5973] - AH9F70r6AA2l

**Comment by mf-code** [comment]
[5973] - pXd9mhibFDPH

**Comment by mf-code** [comment]
[5973] - LZEXJv5ETbqx

**Comment by mf-code** [comment]
[5973] - kD0dCThwxuUr

**Comment by mf-code** [comment]
[5973] - dHxU60FB2StW

**Comment by mf-code** [comment]
[5973] - openjdk8 - Allure report on Netlify: https://5f4cf93bd572722ed170774e--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[5973] - openjdk10 - Allure report on Netlify: https://5f4cfac735dfbb3c91a681c6--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6012] - Mfu9qh8gmmDC

**Comment by mf-code** [comment]
[6012] - MMFT9Xya0fUf

**Comment by mf-code** [comment]
[6012] - F5M52gTGE0rM

**Comment by mf-code** [comment]
[6012] - EEaQN3AUaM3E

**Comment by mf-code** [comment]
[6012] - duFlKC4EnGtz

**Comment by mf-code** [comment]
[6012] - T6fORzP6ITTH

**Comment by mf-code** [comment]
[6012] - s4rvdXIqQ5Kf

**Comment by mf-code** [comment]
[6012] - UHXoYwh6JKgT

**Comment by mf-code** [comment]
[6012] - jF23CM4TMWkd

**Comment by mf-code** [comment]
[6012] - EXWagy6HKoqE

**Comment by mf-code** [comment]
[6012] - zNWyUdmHjNMg

**Comment by mf-code** [comment]
[6012] - NGLuS3KD1PDR

**Comment by mf-code** [comment]
[6012] - y7icRjPmSik9

**Comment by mf-code** [comment]
[6012] - Ayy907oK7PP6

**Comment by mf-code** [comment]
[6012] - hPjvR5T8Q0aH

**Comment by mf-code** [comment]
[6012] - KYYtf0TgaY3l

**Comment by mf-code** [comment]
[6012] - 0XgKiyQcMRbG

**Comment by mf-code** [comment]
[6012] - VEi23BtCfM7z

**Comment by mf-code** [comment]
[6012] - openjdk8 - Allure report on Netlify: https://5f50d8795797540c3faba785--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6012] - openjdk10 - Allure report on Netlify: https://5f50d96dcf00171145276b74--musing-swartz-6fc060.netlify.app

**Review by AlexSofft — COMMENTED** [review]
ok

**Comment by mf-code** [comment]
[6026] - cJb4hTXo3CsE

**Comment by mf-code** [comment]
[6026] - MFspgIy1z4Jt

**Comment by mf-code** [comment]
[6026] - m1tTZ4FvTxow

**Comment by mf-code** [comment]
[6026] - 53JRKXIvWXqT

**Comment by mf-code** [comment]
[6026] - Le4mcvoccPsl

**Comment by mf-code** [comment]
[6026] - lOQcx8IUc3nf

**Comment by mf-code** [comment]
[6026] - h5l4im8VPS6f

**Comment by mf-code** [comment]
[6026] - wwjBZ3JdC1gp

**Comment by mf-code** [comment]
[6026] - UcBgp3SRjncD

**Comment by mf-code** [comment]
[6026] - gpQYn6NfvBDT

**Comment by mf-code** [comment]
[6026] - DkIKTQpp8Dsy

**Comment by mf-code** [comment]
[6026] - uwk8wXYFJuxF

**Comment by mf-code** [comment]
[6026] - ZGjHyBIvwEfU

**Comment by mf-code** [comment]
[6026] - 6xvdMtekByGf

**Comment by mf-code** [comment]
[6026] - LHaRuudze7GN

**Comment by mf-code** [comment]
[6026] - reJjm3ZKlvXJ

**Comment by mf-code** [comment]
[6026] - R6Gi9adMuRuY

**Comment by mf-code** [comment]
[6026] - openjdk8 - Allure report on Netlify: https://5f520150ccdbab9a4d38cd3f--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6026] - openjdk10 - Allure report on Netlify: https://5f5202d3ccdbaba12238c548--musing-swartz-6fc060.netlify.app

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import com.epam.jdi.light.angular.elements.common.Checkbox;
+import com.epam.jdi.light.angular.elements.common.Tooltip;
+import com.epam.jdi.light.elements.common.UIElement;
+import com.epam.jdi.light.elements.composite.Section;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.ui.html.elements.common.Button;
+import com.epam.jdi.light.ui.html.elements.common.TextField;
+
+public class TooltipSection extends Section {
+
```
Comment (gbuh) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java:12]: The TooltipSection should contain several Tooltip elements, not UIElements Buttons, Checkboxes etc.

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    /* Basic tooltip */
```
Comment (gbuh) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java:13]: The comments here indicates that there is a problem with code with they describe.

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    @UI("//tooltip-overview-example/button")
```
Comment (gbuh) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java:14]: same

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    public static Button basicTooltipButton;
+
+    /* Tooltip with a custom position */
```
Comment (gbuh) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java:17]: same

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    @UI("//tooltip-position-example//mat-select")
```
Comment (gbuh) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java:18]: same, all comment below

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/TooltipTests.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+import java.util.Map;
+
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.sections.TooltipSection.*;
+import static io.github.epam.angular.tests.BaseValidationsUtils.duration;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class TooltipTests extends TestsInit {
+
+    @BeforeMethod
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/TooltipTests.java:21]: This method will not run if the previous test case failed.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/TooltipTests.java [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    public void before() {
+        shouldBeLoggedIn();
+        angularPage.shouldBeOpened();
+        if (tooltip.isDisplayed())
+            AngularPage.refresh();
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/common/TooltipTests.java:26]: It's a very time consuming operation.

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/TooltipUnitTests.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Map;
+
+import static org.testng.Assert.assertEquals;
+import static io.github.com.StaticSite.angularPage;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+import static io.github.com.pages.sections.TooltipSection.*;
+
+public class TooltipUnitTests extends TestsInit {
+
+    @BeforeMethod
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/TooltipUnitTests.java:20]: same

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/TooltipUnitTests.java [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+    public void before() {
+        shouldBeLoggedIn();
+        angularPage.shouldBeOpened();
+        if(tooltip.isDisplayed())
+            AngularPage.refresh();
```
Comment (gbuh) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/TooltipUnitTests.java:25]: same

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/TooltipAssert.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+import com.epam.jdi.light.common.JDIAction;
+import com.epam.jdi.light.elements.base.UIBaseElement;
+import org.hamcrest.Matchers;
+
+import static com.epam.jdi.light.asserts.core.SoftAssert.jdiAssert;
+
+public class TooltipAssert extends UIAssert<TooltipAssert, Tooltip> {
+
+    @JDIAction("Assert that '{name}' has '{0}' text")
+    public TooltipAssert assertTooltipText(String text) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/TooltipAssert.java:14]: Inconvenient name of the method.

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/TooltipAssert.java [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+        jdiAssert(element.text(), Matchers.is(text));
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has '{0}' color")
+    public TooltipAssert assertTooltipColor(String color) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/TooltipAssert.java:20]: same

File: jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/TooltipAssert.java [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+        jdiAssert(element.color(), Matchers.is(color));
+        return this;
+    }
+
+    @JDIAction("Assert that '{name}' has '{0}' position relative to '{1}' ")
+    public TooltipAssert assertTooltipPosition(Tooltip.Position position, UIBaseElement<?> e) {
```
Comment (gbuh) [inline jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/TooltipAssert.java:26]: same

File: jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+    /* Tooltip that demonstrates auto-hiding */
+    @UI("//tooltip-auto-hide-example/div")
+    public static UIElement autoHideTooltipContainer;
+
+    @UI("//tooltip-auto-hide-example//button")
+    public static Button autoHideTooltipButton;
+
+    @UI(".mat-tooltip")
+    public static Tooltip tooltip;
```
Comment (gbuh) [inline jdi-light-angular-tests/src/main/java/io/github/com/pages/sections/TooltipSection.java:70]: What exactly "Tooltip" does mean?
In my opinion there are not only pop-up. Pop-up is part of the Angular Tooltip element.
Tooltip must contains  tooltip trigger elements as well.

**Comment by mf-code** [comment]
[6051] - ePvT19pYzZmX

**Comment by mf-code** [comment]
[6051] - XsnGISQkfd0H

**Comment by mf-code** [comment]
[6051] - S5iD3B4mjDJm

**Comment by mf-code** [comment]
[6051] - 1wRFwsRW3usk

**Comment by mf-code** [comment]
[6051] - 2gZwlRLbIo3E

**Comment by mf-code** [comment]
[6051] - JoKBzFPQYUHi

**Comment by mf-code** [comment]
[6051] - EJXej0kmjWmr

**Comment by mf-code** [comment]
[6051] - MiuYOLIVEwXY

**Comment by mf-code** [comment]
[6051] - g5GYhXrKTROn

**Comment by mf-code** [comment]
[6051] - Tsn6cSIPa7Te

**Comment by mf-code** [comment]
[6051] - QVqyjvbAPuxa

**Comment by mf-code** [comment]
[6051] - TIYXFz0Jhkam

**Comment by mf-code** [comment]
[6051] - 0ostqXI0grLb

**Comment by mf-code** [comment]
[6051] - T6UqJpmbGiNL

**Comment by mf-code** [comment]
[6051] - 4cj9Sp04WIui

**Comment by mf-code** [comment]
[6051] - SCaz9USHBOMs

**Comment by mf-code** [comment]
[6051] - 1Pd9ha5agbvu

**Comment by mf-code** [comment]
[6051] - l1r06Vajxmam

**Comment by mf-code** [comment]
[6051] - openjdk8 - Allure report on Netlify: https://5f57c0cccf096250ebc1a5cd--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6051] - openjdk10 - Allure report on Netlify: https://5f57c1b5bed3c1011d00368b--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6093] - L2fCmZOi96KH

**Comment by mf-code** [comment]
[6093] - gFHXqEtgw0JY

**Comment by mf-code** [comment]
[6093] - 4a9B1DfUluKJ

**Comment by mf-code** [comment]
[6093] - aTtFXht73PkG

**Comment by mf-code** [comment]
[6093] - TnX0ZpJyf87o

**Comment by mf-code** [comment]
[6093] - ImmhexdnlAke

**Comment by mf-code** [comment]
[6093] - LEulAJnJUehc

**Comment by mf-code** [comment]
[6093] - 07GBu62HC3zU

**Comment by mf-code** [comment]
[6093] - fTJG5O1N38Xf

**Comment by mf-code** [comment]
[6093] - BjLO1IMrx95N

**Comment by mf-code** [comment]
[6093] - sBqpWCQsfpIi

**Comment by mf-code** [comment]
[6093] - 6lkotdEe94NO

**Comment by mf-code** [comment]
[6093] - uKMs02xxMbw0

**Comment by mf-code** [comment]
[6093] - ABzrekDbIss9

**Comment by mf-code** [comment]
[6093] - n8vJZnFcvO7P

**Comment by mf-code** [comment]
[6093] - XQXQW419QRuI

**Comment by mf-code** [comment]
[6093] - 2oAUJkJ8Vexz

**Comment by mf-code** [comment]
[6093] - 0PaHekdFI6kO

**Comment by mf-code** [comment]
[6093] - openjdk8 - Allure report on Netlify: https://5f5f2facda13e39cb3ccbed8--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6093] - openjdk10 - Allure report on Netlify: https://5f5f3073da13e39b2cccd47c--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6098] - RLVyzhkEzZxm

**Comment by mf-code** [comment]
[6098] - ap2zircF4Ehc

**Comment by mf-code** [comment]
[6098] - 6YQJjyGNOitL

**Comment by mf-code** [comment]
[6098] - uS73no0R1Djv

**Comment by mf-code** [comment]
[6098] - gU0lAJtGXvUz

**Comment by mf-code** [comment]
[6098] - Pxj6qgLotX5q

**Comment by mf-code** [comment]
[6098] - r57CvwxYO7Ik

**Comment by mf-code** [comment]
[6098] - ycJblz4ENnDY

**Comment by mf-code** [comment]
[6098] - qdUJxRRijyp1

**Comment by mf-code** [comment]
[6098] - M7xia0rhY2Cs

**Comment by mf-code** [comment]
[6098] - qEEB1RAhgq3w

**Comment by mf-code** [comment]
[6098] - qrUSmTdKA3TT

**Comment by mf-code** [comment]
[6098] - qFLxJZ2ANybX

**Comment by mf-code** [comment]
[6098] - zR7kDV4DSUHx

**Comment by mf-code** [comment]
[6098] - HbSK51ORcO9b

**Comment by mf-code** [comment]
[6098] - n4qLBh3EOy3t

**Comment by mf-code** [comment]
[6098] - 5f1wpvpXTqLm

**Comment by mf-code** [comment]
[6098] - DCypceaqhvpK

**Comment by mf-code** [comment]
[6098] - openjdk8 - Allure report on Netlify: https://5f5f81c02de74200ead94c76--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6098] - openjdk10 - Allure report on Netlify: https://5f5f827839d0b100dd0f004d--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6098] - J8Lt2hVqWEMJ

**Comment by mf-code** [comment]
[6098] - MWEo6fa9MIUm

**Comment by mf-code** [comment]
[6098] - XU3SAXzo0J7l

**Comment by mf-code** [comment]
[6098] - UzRRyRP4XxyV

**Comment by mf-code** [comment]
[6098] - 0BBoV6E6sznC

**Comment by mf-code** [comment]
[6098] - F0rzcqNKQ3Sq

**Comment by mf-code** [comment]
[6098] - o4dDqLtPLLgt

**Comment by mf-code** [comment]
[6098] - FEq2zDHdZIBH

**Comment by mf-code** [comment]
[6098] - Dhe1E3PfGLX3

**Comment by mf-code** [comment]
[6098] - vF3WOdlXTTr5

**Comment by mf-code** [comment]
[6098] - HqKU0VH1vSSQ

**Comment by mf-code** [comment]
[6098] - CX4VOLJwgpra

**Comment by mf-code** [comment]
[6098] - 6ncyBKAuqL8o

**Comment by mf-code** [comment]
[6098] - i1ZUZuIKZ12Z

**Comment by mf-code** [comment]
[6098] - vexmngEKwm7R

**Comment by mf-code** [comment]
[6098] - jY7C5hFahjfw

**Comment by mf-code** [comment]
[6098] - LebVnbdLOTnC

**Comment by mf-code** [comment]
[6098] - openjdk8 - Allure report on Netlify: https://5f5fc0c3367dcb4bbe7c5993--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6098] - openjdk10 - Allure report on Netlify: https://5f5fc18413fd863974282fd2--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6103] - S5ohy2cj6tzm

**Comment by mf-code** [comment]
[6103] - vIqi3VyV5Gir

**Comment by mf-code** [comment]
[6103] - sipHKa3OVDKr

**Comment by mf-code** [comment]
[6103] - xVK0KVMP2wbr

**Comment by mf-code** [comment]
[6103] - rFQRariyhcIB

**Comment by mf-code** [comment]
[6103] - d4ijpsoG34R7

**Comment by mf-code** [comment]
[6103] - MXEpsfK7V0BY

**Comment by mf-code** [comment]
[6103] - 9WeQag6APQZO

**Comment by mf-code** [comment]
[6103] - xG5wZ69rPpDc

**Comment by mf-code** [comment]
[6103] - KoeRBI049jJO

**Comment by mf-code** [comment]
[6103] - EA104IdVwTAR

**Comment by mf-code** [comment]
[6103] - 0NNSCj5zlTeW

**Comment by mf-code** [comment]
[6103] - 0sXV2rJMOj1E

**Comment by mf-code** [comment]
[6103] - gkx3N0QBS5Qf

**Comment by mf-code** [comment]
[6103] - psmTRuBcTkPv

**Comment by mf-code** [comment]
[6103] - z4WR9nUzgvh8

**Comment by mf-code** [comment]
[6103] - fM5sAKJTMPdv

**Comment by mf-code** [comment]
[6103] - SOFEXpbgRPog

**Comment by mf-code** [comment]
[6103] - openjdk8 - Allure report on Netlify: https://5f6087787d3553ba97d2a8b6--musing-swartz-6fc060.netlify.app

**Comment by mf-code** [comment]
[6103] - openjdk10 - Allure report on Netlify: https://5f6088732b54f1c4ef52ae73--musing-swartz-6fc060.netlify.app

**Pull request merged**